.github/workflows/docker-image.yml

```
@@ -0,0 +1,89 @@
+          with:
+            go-version: '1.23'
+        
+        - name: Build project
+          run: |
+            cd src/product-catalog
+            go mod download
+            go build -o product-catalog-service main.go
+
+        - name: Run unit tests
+          run: go test src/product-catalog/...
+        
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)